src/test/javascript/spec/component/text-cluster-statistics/text-cluster-statistics.component.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { ComponentFixture, TestBed, getTestBed } from '@angular/core/testing';
+import { ArtemisTestModule } from '../../test.module';
+import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.service';
+import { LocalStorageService, SessionStorageService } from 'ngx-webstorage';
+import { MockPipe, MockComponent } from 'ng-mocks';
+import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
+import { ClusterStatisticsComponent } from 'app/exercises/text/manage/cluster-statistics/cluster-statistics.component';
+import { MockTranslateService } from '../../helpers/mocks/service/mock-translate.service';
+import { TranslateService } from '@ngx-translate/core';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { ProgressBarComponent } from 'app/shared/dashboards/tutor-participation-graph/progress-bar/progress-bar.component';
+
+describe('TextClusterStatisticsComponent', () => {
+    let fixture: ComponentFixture<ClusterStatisticsComponent>;
+    let component: ClusterStatisticsComponent;
+    let httpMock: HttpTestingController;
+    let compiled: any;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule, HttpClientTestingModule],
+            declarations: [ClusterStatisticsComponent, MockPipe(ArtemisTranslatePipe), MockComponent(ProgressBarComponent)],
+            providers: [
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: TranslateService, useClass: MockTranslateService },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(ClusterStatisticsComponent);
+                component = fixture.componentInstance;
+                compiled = fixture.debugElement.nativeElement;
+                fixture.detectChanges();
+            });
+        httpMock = getTestBed().get(HttpTestingController);
+    });
+
+    it('should initialize', () => {
+        fixture.detectChanges();
+        expect(component).toBeTruthy();
+    });
```

Comment:
[nit] I am not sure this is necessary, as if component is `undefined` after configuring test module all other tests will fail, right?

Reply:
Not 100% sure either. I have seen this 'pattern' applied all along other test specs, so re-applied the pattern here 😄 . I would assume if for some X reason it is not initialized, we wouldn't be able to infer directly that all tests fail because of initialization failure, so might make debugging easier. 🤔